lib/routes/shopify/apps/search.ts

```diff
@@ -0,0 +1,76 @@
+import { Data, DataItem, Route } from '@/types';
+import type { Context } from 'hono';
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { baseURL } from './const';
+
+export const route: Route = {
+    path: '/apps/search/:q',
+    example: '/shopify/apps/search/flow',
+    parameters: { q: '需要搜索的 App' },
+    name: 'Shopify App store search',
+    maintainers: ['PrintNow'],
+    handler,
+};
+
+async function handler(ctx?: Context): Promise<Data> {
+    if (!ctx) {
+        throw new Error('未传入 ctx');
+    }
+
+    const { q = '' } = ctx.req.param();
+    const response = got.get(`${baseURL}/search`, {
+        searchParams: {
+            q,
+        },
+        responseType: 'text',
+        headers: {
+            'accept': 'text/html, application/xhtml+xml',
+            'accept-language': 'en-US;q=0.9',
+            'turbo-frame': 'search_page',
+            'referer': baseURL,
+            'dnt': '1',
+            'user-agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
```

Comment:
RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/263d6b818ea551e9a498207ba1f0a912b6370224/lib/config.ts#L344) by default.